### PR TITLE
grex: init at 1.0.0

### PR DESCRIPTION
**Pull request by Br1ght0ne**
###### Motivation for this change
https://github.com/pemistahl/grex/releases/tag/v1.0.0
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---
This change is [Reviewable](https://reviewable.io/reviews/nixos/nixpkgs/76539)

### CODE REVIEW

File: pkgs/tools/text/grex/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  };
+
+  cargoSha256 = "1l6vwdib0wxag8hvsjhqmy2n0ri15qf9drnimppcg3qcpz3mlga0";
+
+  meta = with lib; {
+    description = "Generate regular expressions from user-provided test cases";
+    homepage = "https://github.com/pemistahl/grex";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ filalex77 ];
+    platforms = platforms.all;
```
Comment (jonringer) [inline pkgs/tools/text/grex/default.nix]: just inherit platforms from buildRustPackage, otherwise LGTM
```suggestion
```

**Comment by jonringer** [comment]
please squash commits:

```
git reset HEAD^
git add pkgs/
git commit --amend --no-edit
git push .. .. --force
```

**Comment by filalex77** [comment]
@jonringer Done.

**Comment by jonringer** [comment]
there was some activity regarding rust packages, please update 

https://github.com/NixOS/nixpkgs/issues/79975

**Comment by bhipple** [comment]
@GrahamcOfBorg build grex

File: pkgs/tools/text/grex/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  version = "0.3.2";
+
+  src = fetchFromGitHub {
+    owner = "pemistahl";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "141bf7lhclpxdgjzvznq7ns4kplxbfkjq03rx91b95dch6c6bfl4";
+  };
+
+  cargoSha256 = "1l6vwdib0wxag8hvsjhqmy2n0ri15qf9drnimppcg3qcpz3mlga0";
```
Comment (bhipple) [inline pkgs/tools/text/grex/default.nix]: Rebase onto origin/master, then:

```suggestion
  cargoSha256 = "02chy973na34ncc648v5hfrh1l48qjpbnx3qnkdig7k4pz29ynyj";
```

**Review by bhipple — CHANGES_REQUESTED** [review]

**Comment by jonringer** [comment]
closing in favor of other PR

**Pull request closed without merging**